app/models/degree.rb

```diff
@@ -6,15 +6,9 @@ class Degree < ApplicationRecord
     type_of_degree: 20
   }.freeze
 
-  validates :type_of_degree, presence: true
-  validates :subject, presence: true
-  validates :institution, presence: true
-  validates :class_of_degree, presence: true
-  validates :year, presence: true
+  MAX_LENGTHS.each { |field, max| validates field, length: { maximum: max } }
 
-  validates :year, numericality: { greater_than: 1899, less_than_or_equal_to: DateTime.now.year }
+  validates :type_of_degree, :subject, :institution, :class_of_degree, :year, presence: true
```

Comment:
Please move presence validations to the top so they will be the first error per field, which the form builder displays